### ISSUE TITLE
Throwing exceptions when trying to get layouts.

### ISSUE THREAD

**Issue by YaruKatsaros** [OndrejNepozitek/ProceduralLevelGenerator#41]
When trying to use the DLLs included on the latest release on a new project, it throws a DllNotFoundException `(GeneralAlgorithms.Algorithms.Common.BoostWrapper.IsPlanar(Int32[] edges, Int32 edgesCount, Int32 verticesCount)` if Boostwrapper isn't found (although the installation tutorial indicates it isn't necessary), and if the DLL is added to the bin folder (since Virtual Studio refuses to add it as a dependency), it throws a NullReferenceException(`GeneralAlgorithms.Algorithms.Common.CollectionExtensions.GetWeightedRandom[T](IList`1 elements, Func`2 weightSelector, Random random)`, in both cases when trying to call `layoutGenerator.GetLayout`

Just in case it helps, this is my code:
```
protected override void Initialize()
        {
            base.Initialize();
            Utils.CenterWindow(this, graphics);
            IsMouseVisible = true;

            var mapDescription = new MapDescription<int>();
            mapDescription.AddRoom(0);
            mapDescription.AddRoom(1);
            mapDescription.AddRoom(2);
            mapDescription.AddRoom(3);

            mapDescription.AddPassage(0, 1);
            mapDescription.AddPassage(0, 3);
            mapDescription.AddPassage(1, 2);
            mapDescription.AddPassage(2, 3);

            var doorMode = new OverlapMode(1, 1);
            var squareRoom = new RoomDescription(GridPolygon.GetSquare(8), doorMode);
            var rectangleRoom = new RoomDescription(GridPolygon.GetRectangle(6, 10), doorMode);
            var corridor = new RoomDescription(GridPolygon.GetSquare(1), new OverlapMode(1, 1));

            mapDescription.AddRoomShapes(squareRoom);
            mapDescription.AddRoomShapes(rectangleRoom);
            mapDescription.AddCorridorShapes(corridor);
            mapDescription.SetWithCorridors(true, new List<int>() { 1 });

            var layoutGenerator = LayoutGeneratorFactory.GetChainBasedGeneratorWithCorridors<int>(new List<int>() { 1 });
            var generatedLayouts = layoutGenerator.GetLayouts(mapDescription, 1);

        }
```

**Comment by OndrejNepozitek** [comment]
Unfortunately, there are situations where not all preconditions are checked and the algorithm ends up throwing a generic exception only after something goes wrong (there's an issue created to improve this). 

From what I can see from your code, I'd say that there is a problem with your definition of the corridor room. You create a 1x1 square and then use the `new OverlapMode(1, 1)` door mode. By using that door mode, you tell the algorithm that you want all doors to be at least one "tile" from corners of the room shape. And as you use 1x1 square, there are no door positions that match your requirements. I'd suggest you to try using `var corridor = new RoomDescription(GridPolygon.GetSquare(1), new OverlapMode(1, 0));` (note the 0 instead of 1). It will allow 4 door positions for that corridor room and should solve the issue.

This is absolutely something that should be handled by an exception with some useful message.

**Comment by YaruKatsaros** [comment]
Thank you for the quick response! Alas, the problem persists: I've tried setting the Overlapmode to (1,0), and to remove the corridors and change the generator to a `LayoutGeneratorFactory.GetDefaultChainBasedGenerator<int>()`, and even just copypasting the example shown [here](https://github.com/OndrejNepozitek/ProceduralLevelGenerator/blob/master/Sandbox/Examples/BasicsExample.cs) but it keeps throwing a NullReferenceException on the same line.

**Comment by OndrejNepozitek** [comment]
I think that I've found the problem. When implementing the library, it took some time to solve all the big bugs in the core of the algorithm. And as the algorithm is stochastic, it sometimes happened that a bug showed up only when you were quite unlucky. To fix such bugs, you have to fix the seed of the random numbers generator so that you always get the same result and can debug the issue. So to prevent me from accidentally not setting the correct seed for the generator, I removed the default instance of the random generator. And that's exactly the problem you're encountering - a missing instance of the Random class.

I uploaded a newer version of the binaries that should have this issued fixed. But before downloading the newer version, please try a quick manual fix to make sure that it was the only problem. Before running the layout generator, please inject the random numbers generator like this:
```csharp
layoutGenerator.InjectRandomGenerator(new Random());
```

**Comment by YaruKatsaros** [comment]
Indeed, that does solve it!
Thank you very much for helping and for all your work on the generator!

**Comment by OndrejNepozitek** [comment]
I'm glad that it worked! If you have any problems in the future, feel free to open another issue.

I'd also love to get any feedback on the library - either on the features of the algorithm or how usable it's for a programmer. So if you find anything that you feel is not very intuitive or if you miss a feature, just please let me know. There are quite a few places that I already know are not perfect, but it'd be great to hear from other people, too.
